<a href="https://colab.research.google.com/github/sovikapoudel/Data-Visualiation-Spring-2022/blob/main/Quiz_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1><center>Data Visualization Spring 2022</center></h1>

---

###Assignment: 6
Name: Sovika Poudel <br>
Year: Junior <br>
Course: CPSMA-4313 Data Processing Visualization 
### For this part of Assignment I am going to load the libraries, gather dataset and load them here in this notebook and copy all my work from previous assignment for the better and easier reference.

In [ ]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Gathering Data

In [ ]:
%%bigquery --project data-vis-assignment-338801
SELECT *
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  lIMIT 5



trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19

[5 rows x 9 columns]

What was the second most popular starting station?

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT start_station_name, COUNT(start_station_id) as total_starting_point
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  GROUP BY start_station_name
  Order BY total_starting_point DESC ###this will give the most popular on the top so we can see 2nd most popular in 2nd row
  LIMIT 5

start_station_name  total_starting_point
0      21st & Speedway @PCL                 72799
1      Riverside @ S. Lamar                 40635
2  City Hall / Lavaca & 2nd                 36520
3            2nd & Congress                 35307
4      Rainey St @ Cummings                 34758

So the second most popular starting station is Riverside @ S. Lamar	with 40635 counts

How many trips lasted over an hour and were a round trip (started and stopped at the same station)?

At first, for this part I am just showing the list in order and confirming if the query is running correctly.

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT start_station_name, end_station_name, duration_minutes as time_travelled
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  WHERE  start_station_name = end_station_name and  duration_minutes >60
  LIMIT 100

start_station_name  ... time_travelled
0                   Toomey Rd @ South Lamar  ...             82
1                          Waller & 6th St.  ...             76
2                          Waller & 6th St.  ...             75
3                          Waller & 6th St.  ...             74
4                          Waller & 6th St.  ...             73
..                                      ...  ...            ...
95                  Toomey Rd @ South Lamar  ...             88
96                  Toomey Rd @ South Lamar  ...             73
97                  Toomey Rd @ South Lamar  ...             72
98                         Waller & 6th St.  ...             61
99  Capital Metro HQ - East 5th at Broadway  ...            192

[100 rows x 3 columns]

Now, for the total numbers of loop that were over 60 minutes I am using a clause within a 'WHERE' clause

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT COUNT(*) as No_of_Overthehour_loops
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  WHERE  start_station_name = end_station_name and  duration_minutes >60
  
  # Order by time_travelled DESC ### this line of code is not working correctly on this part. so commenting this out for now.
 

No_of_Overthehour_loops
0                    54795

##Assignment 6 begins from here


###Task to do!

Examine the average trip on Austin bike share grouping by property_type column from the bikeshare_stations table. Is average time on trips starting from 'parkland' longer than from 'sidewalks'? To answer this question you must join the bikeshare_trips and bikeshare_stations on the start_station_name and the name from their respective tables.

SQL Joins

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_stations`
WHERE property_type != "NONE"
Limit 10

station_id  ...             modified_date
0        2502  ... 2021-01-04 12:00:00+00:00
1        3687  ... 2021-01-04 12:00:00+00:00
2        3684  ... 2021-01-04 12:00:00+00:00
3        2572  ... 2021-01-04 12:00:00+00:00
4        2575  ... 2021-01-04 12:00:00+00:00
5        2574  ... 2021-01-04 12:00:00+00:00
6        2707  ... 2021-01-04 12:00:00+00:00
7        1111  ... 2021-11-04 02:12:00+00:00
8          11  ... 2021-11-04 02:24:00+00:00
9        4699  ... 2021-07-27 08:25:00+00:00

[10 rows x 14 columns]

First, let's gather the data about the property type and it's count.

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT property_type as Type, COUNT(property_type ) as Num
FROM `bigquery-public-data.austin_bikeshare.bikeshare_stations` 
WHERE property_type != " " ##There's one blank entry in the dataset under property_type so I am using the where clause to remove that from my output
group by  property_type
LIMIT 20 #just checking if this pulls out the all the informations, we only have total of 6 entries for this query though

Type  Num
0              parkland    6
1              sidewalk   28
2          paid_parking   19
3    nonmetered_parking    8
4  undetermined_parking   19

Now, we have got the total numbers of each property types. Let's join these both datasets for our next part.

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT trip.start_station_id, trip.start_station_name, station.property_type, trip.duration_minutes as Trip_Duration
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as trip INNER JOIN `bigquery-public-data.austin_bikeshare.bikeshare_stations` as station 
ON trip.start_station_name=station.name AND station.property_type != "None" AND station.property_type != " "
LIMIT 20

start_station_id  ... Trip_Duration
0               2823  ...            16
1               2823  ...            23
2               2823  ...             4
3               2823  ...            10
4               2568  ...             3
5               2568  ...            11
6               3291  ...            20
7               3291  ...            35
8               3291  ...            19
9               2568  ...            27
10              3291  ...             7
11              2823  ...            13
12              3293  ...            32
13              2568  ...             0
14              3293  ...            31
15              2568  ...             0
16              1007  ...            17
17              2568  ...             3
18              1007  ...            19
19              1007  ...             0

[20 rows x 4 columns]

Now, Let's find out the average duration time of each property type by grouping them together by property_type

In [ ]:
%%bigquery --project data-vis-assignment-338801 
SELECT station.property_type, AVG(trip.duration_minutes) as Average_Trip_Duration
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as trip INNER JOIN `bigquery-public-data.austin_bikeshare.bikeshare_stations` as station 
ON trip.start_station_name=station.name AND station.property_type != "None" AND station.property_type != " "
GROUP BY station.property_type
LIMIT 20



property_type  Average_Trip_Duration
0  undetermined_parking              23.098028
1              sidewalk              29.263569
2    nonmetered_parking              49.046326
3          paid_parking              25.852535
4              parkland              39.375957

Pfttt!! This was a hardest one so far! But got a chance to learn better. Not gonna lie but it took me way more than an hour.